.gitlab-ci.yml

```diff
@@ -56,18 +56,28 @@ validate-log-integrations:
 trigger-agent-build:
   stage: trigger
   image: $VALIDATE_AGENT_BUILD
+  # Exclude release branches without dependency change
+  rules:
+    - if: ($CI_COMMIT_BRANCH =~ /^7.[0-9]{2}.x/)
+      changes:
+        - datadog_checks_base/datadog_checks/base/data/agent_requirements.in
+      when: always
+    - if: ($CI_COMMIT_BRANCH !~ /^7.[0-9]{2}.x/)
+      when: always
```

Comment:
The `trigger-agent-build`  job is ignored on release branches (branches looking like `7.YY.x`) when `agent_requirements.in` is not changed.
So, cherry-picked commit on release branch most of the time WON'T trigger agent build

Reply:
I don't know know if `changes:` works here: https://gitlab.com/gitlab-org/gitlab/-/issues/11427

Reply:
This issue implies the job will be triggered the first time a release branch is pushed. The first time we push the branch, we're based on master, so the agent build won't be trigger (https://github.com/DataDog/integrations-core/pull/10842/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R75)

The rest of the time, the branch has already been pushed once, so the issue shouldn't apply

Reply:
Ok nice, I see, so we're only using `only:changes` on the release branch so this part should work since it's per commit